### DISCUSSION TITLE
Required and Optional aren't fine-grained enough for dependencies

### DISCUSSION THREAD

**Discussion by SoniEx2** [modrinth/.github#76] [Feature requests]
### Is your feature request related to a problem? Please describe.

Our mod doesn't technically require fabric API, but it'll have limited functionality like not being able to show the generated friend codes. This is because it relies on the lang files for that, and those are only loaded when using fabric API. (The rest of the mod, in particular being able to connect to friend codes, works fine. Obviously lang files are pretty important for having human-readable text on the UI, so arguably we should just require fabric API already, but we digress.)

### Describe the solution you'd like

We'd like the ability to have a "Recommends" or "Soft-Required", with the description of "major functionality will be missing without this dependency". Similar to how fabric.mod.json does it.

### Describe alternatives you've considered

We could just mark it straight-up required... Ah well.

### Additional context

N/A

**Comment by Madis0** [comment]
> Obviously lang files are pretty important for having human-readable text on the UI, so arguably we should just require fabric API already, but we digress.

It does look like your mod should just require the API or bundle the parts it needs. Why overcomplicate the definition when it would clearly appear broken to users without it?

**Comment by CodexNotFound** [comment]
I definitely see some use case for this as well.

Example 1: There are plenty of mods that are not needed to run your mod but your mod has explicit support for it. Think of mods like AntiqueAtlas, running it on an otherwise vanilla client is great, but it has special assets for displaying biomes from other mods.

Example 2: A mod is designed with the goal to hook into another mod (eg. Crafting recipes using Create), but has "backwards" compatibility for when a user doesn't have Create. It doesn't really impact the gameplay too much if you don't have it, but it's not the "intended" way.

**Comment by quat1024** [comment]
All these use cases still feel like they fit comfortably in Optional. It's an optional dependency, it's not required.

**Comment by triphora** [comment]
This has been implemented in API v3 (not yet ready for public use, but still available on production) as four booleans - `client_and_server`, `client_only`, `server_only`, `singleplayer`.

**Reply by SoniEx2** [reply]
huh? this is about dependencies, not sidedness. (tho as far as sidedness goes, that seems like a huge improvement yeah.)

**Reply by triphora** [reply]
Oops. My bad.
